### ISSUE TITLE
Graphql Authorization via application properties

### ISSUE THREAD

**Issue by debu999** [quarkusio/quarkus#27355]
@sberyozkin @phillip-kruger 
For rest APIs we have option to add authorization based on endpoints. 
![Screenshot 2022-08-18 at 2 46 45 PM](https://user-images.githubusercontent.com/1910577/185358966-80f51d8c-734f-4270-97ba-b3a832d8b8db.png)
https://quarkus.io/guides/security-authorization
How can we have something similar for graphql endpoints. Currently the only option is to add @RolesAllowed annotation on top of the graphql APIs.

For example i have 2 query and 2 mutations for Case and Audit and i need Case to be accessible if role is CaseUser and Audit role is for AuditUser. 
how to specify this in application.properties similar to rest apis configuration for security.

**Comment by phillip-kruger** [comment]
You can add the @RolesAllowed on the method (so the queries). So for example:
```
@Query
@RolesAllowed("CaseUser")
public Case caseQuery(){
///
}

@Mutation
@RolesAllowed("CaseUser")
public Case caseMutation(){
///
}


@Query
@RolesAllowed("AuditUser")
public Audit auditQuery(){
///
}

@Mutation
@RolesAllowed("AuditUser")
public Audit auditMutation(){
///
}
```

**Comment by debu999** [comment]
@phillip-kruger Yes that works like a charm. But same is also provided in Rest Controller Resources.

What i wanted to know is can we have it in application properties 
This allows us to have common place to maintain the roles for queries and not to have several @RolesAllowed in each query if they are specific. 

This might need a enhancement but definitely can be a good thing. As a lot comes via properties in Quarkus.

so i can state something like below
```
quarkus.graphql.auth.policy.role-policy1.roles-allowed=AuditUser                     
quarkus.graphql.auth.permission.role1.queries=auditQuery,auditManagement  
quarkus.graphql.auth.permission.role1.mutations=auditMutation,auditDelete        
quarkus.graphql.auth.permission.role1.policy=role-policy1

quarkus.graphql.auth.policy.role-policy2.roles-allowed=CaseUser                     
quarkus.graphql.auth.permission.role2.queries=caseQuery,caseManagement  
quarkus.graphql.auth.permission.role2.mutations=caseMutation,caseDelete,escalateCase   
quarkus.graphql.auth.permission.role2.policy=role-policy2
```

**Comment by phillip-kruger** [comment]
No that can not be done at the moment. We can maybe looking at adding this feature, but we will have to do this on GraphQL level, as there is only one endpoint (`/graphql`)

**Comment by debu999** [comment]
> No that can not be done at the moment. We can maybe looking at adding this feature, but we will have to do this on GraphQL level, as there is only one endpoint (`/graphql`)
> 

Yes that's the thing with graphql. Looking forward to this in future

**Comment by Jamel-jun** [comment]
I have a question. If I want to control a model based on roles or permissions, for example, admin will display the user's plaintext password, while auditor will not display the user's password, the user displays the encrypted password. How should this be achieved?

**Comment by phillip-kruger** [comment]
This can not be done at the moment. The model is created on build time, and this would require runtime knowledge

**Comment by Jamel-jun** [comment]
I read a document that can meet the requirements of roles [Secure serialization](https://quarkus.io/guides/security-overview-concept#secure-serialization).
It seems that permission processing is not supported.
